packages/coreutils/src/interfaces.ts

```diff
@@ -98,6 +100,90 @@ export interface IDataConnector<T, U = T, V = string> {
   save(id: V, value: U): Promise<any>;
 }
 
+/**
+ * A tracker that tracks instances of a generic objects.
+ *
+ * @typeparam T - The type of object being tracked.
+ */
+export interface IInstanceTracker<T extends IObservableDisposable>
+  extends IDisposable {
+  /**
+   * A signal emitted when an instance is added.
+   *
+   * ####
+   * This signal does not emit if an instance is added using `inject()`.
+   */
+  readonly added: ISignal<this, T>;
+
+  /**
+   * The current object instance.
+   */
+  readonly current: T | null;
```

Comment:
Clever, I like this. It's similar to how the `ISignal/Signal` interface handles emissions.

Reply:
That's where I got the idea!

Reply:
`Poll` and `IPoll` are a variant of this pattern as well, where none of the methods that can modify a poll's internal state are defined in `IPoll`, it is strictly consumer-only, no 'start()`, 'stop()`, 'refresh()`, etc.